### ISSUE TITLE
Code of Conduct?

### ISSUE THREAD

**Issue by ChALkeR** [qmlweb/qmlweb#278]
Several questions here.

 1. Have we reached the state when we should adopt a Code of Conduct? I expect to have more eyes on this, and I don't feel like forcing some random one.
 2. Which one should we adopt?

For reference, KDE has one: https://www.kde.org/code-of-conduct/, but it's specifically tied to KDE all over it.

Other examples:

* https://github.com/nodejs/node/blob/master/CODE_OF_CONDUCT.md
* https://www.rust-lang.org/conduct.html
* http://contributor-covenant.org/version/1/3/0/
* http://contributor-covenant.org/version/1/4/
* probably many more that I have not mentioned here.

/cc @qmlweb/collaborators

**Comment by JoshuaKolden** [comment]
It depends on what your goal is.  I generally find "Code of Conduct" documents are widely ignored.  People conduct themselves however they like.  I've seen groups cite a Code of Conduct to help prevent flame wars, but I've seen groups cite a Code of Conduct even more often to silence dissenting opinions.  For the most part I think they are completely useless self-righteous drivel as likely to be abused as to be helpful, but there is an exception.

When a Code of Conduct (CoC) is used to limit the power of those who have it in favor of being more open and enabling to new members such as when a CoC is used for **enabling rather than disabling** to welcome contributions from anyone regardless of race, religion, gender, or sexual orientation for example.

So just looking a the "KDE" CoC as an example, I'd say it's particularly bad.  The prescriptions "be considerate", "be respectful", "be collaborative", are very nice and positive sounding but are actually poorly defined, and open to broadly differing subjective opinion.  In response to someone crossing this vague line there is only one suggested remedy, "suspended access". 

**As vague as it all is, what little guidance it could provide is lost with the sweeping declaration "Leaders of any group, such as moderators of mailing lists, IRC channels, forums, etc., will exercise the right...".**

This statement alone basically nullifies any other effect of the document by saying in effect "moderators can do what they want".  Well no shit KDE!  Of course moderators can do what they want.

Leaders and moderators already have that power without a CoC, so KDE's CoC has no effect on anything.  Perhaps it very slightly escalates the power of moderators by giving them a document to point to when someone complaints about losing access when a moderator used their unconstrained power to block them.

To me a CoC should have exactly the opposite power.  It should provided guidance to the moderators that specifically limits their power. It must not be vague, and it must have teeth.  It should say what behavior is not allowed, but remain silent on what is allowed, since everything that is not explicitly prohibited is explicitly allowed so as to avoid a 'chilling effect'[1].  And it should be clear what the reproductions are of disobeying the CoC.  This structure of CoC encourages communication, dissenting opinions, unexpected contributions, and new ideas. 

However, this use of a CoC in the open source community is exceedingly rare. Since why wold a "leader" or "moderator" voluntarily limit their own power?  Nevertheless, it's the only form of CoC that makes any sense to me.  

On the other hand KDE's CoC could be perfectly reasonable for a Private Club. Private Clubs are not concerned with innovation or discovery, they are concerned with peace and ease.  The kind of thing you have when you are surrounded by like minded people.  Private Clubs like this are generally not very politically correct (think exclusive gulf clubs, etc.), but I actually think they have just as much right to exist as anything else.  It's reasonable to have a place where you can relax without a constant debate about your interests or beliefs.  I just think one should not pretend to be the other.

1: https://en.wikipedia.org/wiki/Chilling_effect

**Comment by akreuzkamp** [comment]
I also think, common sense should be enough. If it would happen, that we'd need to rebuke or even oust someone, because of the CoC, he/she would be just as upset, as if we would oust him/her for "common sense". As @JoshuaKolden said, the only real use of a CoC would be to limit the power of moderators/admins. But I don't see the necessity for that, right now.

**Comment by ChALkeR** [comment]
@akreuzkamp Note: #26 also means adopting a Code of Conduct.

**Comment by akreuzkamp** [comment]
that's true. To clarify: I'm not *against* a CoC, I just see no need for it. :)
